packages/webview_flutter/webview_flutter_platform_interface/lib/src/v4/webview_controller_delegate.dart

```diff
@@ -0,0 +1,264 @@
+
+import 'types/types.dart';
+import 'webview_platform.dart';
+
+/// Interface for a platform implementation of a web view controller.
+///
+/// Platform implementations should extend this class rather than implement it
+/// as `webview_flutter` does not consider newly added methods to be breaking
+/// changes. Extending this class (using `extends`) ensures that the subclass
+/// will get the default implementation, while platform implementations that
+/// `implements` this interface will be broken by newly added
+/// [WebViewCookieManagerDelegate] methods.
+abstract class WebViewControllerDelegate extends PlatformInterface {
+  /// Creates a new [WebViewControllerDelegate]
+  factory WebViewControllerDelegate() {
+    final WebViewControllerDelegate webViewControllerDelegate =
+        WebViewPlatform.instance!.createWebViewControllerDelegate();
+    PlatformInterface.verify(webViewControllerDelegate, _token);
+    return webViewControllerDelegate;
+  }
+
+  /// Used by the platform implementation to create a new
+  /// [WebViewControllerDelegate].
+  ///
+  /// Should only be used by platform implementations because they can't extend
+  /// a class that only contains a factory constructor.
+  @protected
+  WebViewControllerDelegate.implementation() : super(token: _token);
+
+  static final Object _token = Object();
+
+  /// Loads the file located on the specified [absoluteFilePath].
+  ///
+  /// The [absoluteFilePath] parameter should contain the absolute path to the
+  /// file as it is stored on the device. For example:
+  /// `/Users/username/Documents/www/index.html`.
+  ///
+  /// Throws an ArgumentError if the [absoluteFilePath] does not exist.
+  Future<void> loadFile(
+    String absoluteFilePath,
+  ) {
+    throw UnimplementedError(
+        'WebView loadFile is not implemented on the current platform');
+  }
+
+  /// Loads the Flutter asset specified in the pubspec.yaml file.
+  ///
+  /// Throws an ArgumentError if [key] is not part of the specified assets
+  /// in the pubspec.yaml file.
+  Future<void> loadFlutterAsset(
+    String key,
+  ) {
+    throw UnimplementedError(
+        'WebView loadFlutterAsset is not implemented on the current platform');
+  }
+
+  /// Loads the supplied HTML string.
+  ///
+  /// The [baseUrl] parameter is used when resolving relative URLs within the
+  /// HTML string.
+  Future<void> loadHtmlString(
+    String html, {
+    String? baseUrl,
+  }) {
+    throw UnimplementedError(
+        'WebView loadHtmlString is not implemented on the current platform');
+  }
+
+  /// Loads the specified URL.
+  ///
+  /// If `headers` is not null and the URL is an HTTP URL, the key value paris in `headers` will
+  /// be added as key value pairs of HTTP headers for the request.
+  ///
+  /// `url` must not be null.
+  ///
+  /// Throws an ArgumentError if `url` is not a valid URL string.
+  Future<void> loadUrl(
```

Comment:
I don't believe we need this method in the new interface. `loadRequest` should be the main method in the new interface. Unless we want to keep this as a simpler alternative. What are your thoughts @stuartmorgan?

Reply:
Yes, `loadRequest` was intended to be a future-proof replacement for `loadUrl`, so we should remove this.

Reply:
Removed the `loadUrl` method from `WebViewControllerDelegate`.